File: 프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        long answer = Long.MAX_VALUE;
        Arrays.sort(times);
        long st=0, en=(long) n*times[times.length-1];
        while(st<=en) {
            long mid = (st+en)/2;
            long sum = 0;
            for(int i : times) sum += (mid/i);
            if(sum>=n) {
                en=mid-1;
                answer = Math.min(answer,mid);
            } else st=mid+1;
        }
        return answer;
    }
}

File: 프로그래머스/3/43238. 입국심사/입국심사.cpp
#include <bits/stdc++.h>
using namespace std;

long long solution(int n, vector<int> times) {
    long long answer = LLONG_MAX;
    sort(times.begin(),times.end());
    long long st=0, en=1LL*n*times[times.size()-1];
    while(st<=en) {
        long long mid=(st+en)/2;
        long long sum=0;
        for(int i : times) sum+=mid/i;
        if(sum>=n) {
            en=mid-1;
            answer=min(answer,mid);
        } else st=mid+1;
    }
    return answer;
}